sky/execution.py

```diff
@@ -41,6 +41,27 @@
 OptimizeTarget = optimizer.OptimizeTarget
 _MAX_SPOT_JOB_LENGTH = 10
 
+# Error message thrown when sky.exec() receives a string instead of a Dag.
+_EXEC_STRING_AS_DAG_HINT_MESSAGE = """\
+Programmatic API sky.exec() expects a sky.Dag but received a string.
```

Comment:
nit: `_execute()` can also be called by `sky.launch` or `sky.spot_launch`. Would it make sense to write this help str in a call agnostic manner? E.g.

```
_EXEC_STRING_AS_DAG_HINT_MESSAGE = """\
Expected a sky.Dag but received a string.
If you meant to run a command, make it a Task's run command and wrap as a Dag:
  def to_dag(command, gpu=None):
      with sky.Dag() as dag:
          sky.Task(run=command).set_resources(
              sky.Resources(accelerators=gpu))
      return dag

The command can then be run as:
  sky.exec(to_dag(cmd), cluster_name=..., ...) # No GPUs required
  sky.exec(to_dag(cmd, gpu={'V100': 1}), cluster_name=..., ...)
  sky.launch(to_dag(cmd, gpu={'V100': 1}), cluster_name=..., ...)
  sky.spot_launch(to_dag(cmd, gpu='V100:0.5'), cluster_name=..., ...)
""".strip()
```

Of course, the message is clear even if we don't make the help str call agnostic so feel free to not make this change.

Reply:
Good catch! Did the change and re-tested all three callers.  PTAL.